### PR TITLE
Bring with_retries and extra logging into tools.sh's acquisition of dotnet-install

### PR DESCRIPTION
**Pull request by MattGal**
### To double check:

* [x] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/core-eng/tree/master/Documentation/Validation

(This code runs with every linux PR build leg and will break the build if it fails)

### CODE REVIEW

**Review by chcosta — APPROVED** [review]
Looks like a sane port to me

File: eng/common/tools.sh [hunk 1]
```diff
@@ -246,13 +268,22 @@ function GetDotNetInstallScript {
 
     # Use curl if available, otherwise use wget
     if command -v curl > /dev/null; then
+      # first, try directly, if this fails we will retry with verbose logging
       curl "$install_script_url" -sSL --retry 10 --create-dirs -o "$install_script" || {
-        local exit_code=$?
-        Write-PipelineTelemetryError -category 'InitializeToolset' "Failed to acquire dotnet install script (exit code '$exit_code')."
-        ExitWithExitCode $exit_code
+        if command -v openssl &> /dev/null
+        then
```
Comment (chcosta) [inline eng/common/tools.sh:274]: I think our convention here is to do `if blah; then` on one line

Comment (MattGal) [reply eng/common/tools.sh:274]: since it's this way in main, will ignore this.

Comment (chcosta) [reply eng/common/tools.sh:274]: Not sure I agree with that logic.  Looks like you authored this originally in main as well.  My bad for not catching this before I hit Approve.  I'll create a Pr for main.

**Pull request merged**